meson.build

```diff
@@ -9,11 +9,67 @@ project('imgui', 'cpp',
 include_dirs = include_directories('.', 'examples')
 
 sources = ['imgui_demo.cpp', 'imgui_draw.cpp', 'imgui_widgets.cpp', 'imgui.cpp']
+if get_option('opengl3').enabled()
+	sources += 'examples/imgui_impl_opengl3.cpp'
+endif
+if get_option('dx12').enabled()
+	sources += 'examples/imgui_impl_dx12.cpp'
+endif
+if get_option('dx9').enabled()
+	sources += 'examples/imgui_impl_dx9.cpp'
+endif
+if get_option('opengl2').enabled()
+	sources += 'examples/imgui_impl_opengl2.cpp'
+endif
+if get_option('glut').enabled()
+	sources += 'examples/imgui_impl_glut.cpp'
+endif
+if get_option('dx11').enabled()
+	sources += 'examples/imgui_impl_dx11.cpp'
+endif
+if get_option('win32').enabled()
+	sources += 'examples/imgui_impl_win32.cpp'
+endif
+if get_option('sdl').enabled()
+	sources += 'examples/imgui_impl_sdl.cpp'
+endif
+if get_option('glfw').enabled()
+	sources += 'examples/imgui_impl_glfw.cpp'
+endif
+if get_option('dx10').enabled()
+	sources += 'examples/imgui_impl_dx10.cpp'
+endif
+if get_option('allegro5').enabled()
+	sources += 'examples/imgui_impl_allegro5.cpp'
+endif
+if get_option('vulkan').enabled()
+	sources += 'examples/imgui_impl_vulkan.cpp'
+endif
+if get_option('marmalade').enabled()
+	sources += 'examples/imgui_impl_marmalade.cpp'
+endif
+if get_option('osx').enabled()
+	sources += 'examples/imgui_impl_osx.mm'
+endif
+if get_option('metal').enabled()
+	sources += 'examples/imgui_impl_metal.mm'
+endif
 
+# TODO: add other dependencies
+vulkan = dependency('vulkan', required: get_option('vulkan'))
```

Comment:
Please use `vulkan_dep` and so on for consistency with the other dependency variables.
Otherwise, why do we even need separate variable? Why not inline this into the list they are used in below? Or declare it as a list? It's basically `x = dependency('x')`, so having a variable doesn't improve readability.